### PR TITLE
Fixvmsend

### PR DESCRIPTION
**Pull request by ebuchman**
Solves first half of https://github.com/tendermint/tendermint/issues/65 but leaves the matter of CALLing an unknown account unresolved. Presumably we can just create a new vm.Account at https://github.com/tendermint/tendermint/blob/fixvmsend/vm/vm.go#L717 and things will be fine.

### CODE REVIEW

**Comment by ebuchman** [comment]
Other TODO: 
[ ] `CallTx` should make sure code is not empty if address exists

**Comment by jaekwon** [comment]
Maybe if op == CALLCODE it should still fail?  Not sure why you'd want to CALLCODE on an account.
Otherwise sounds good, we (should) just need to CreateAccount as in op == CREATE.

**Comment by ebuchman** [comment]
ya so I guess CALLCODE should fail if the code is empty.

Also noticed errors inside the vm aren't propogating/being logged that well, so will add some debug and probably an event for when a CALL fails

**Comment by ebuchman** [comment]
and its not quite creating an account like in CREATE, since we're given the address already and just need to initialize it into the tree (like if it had received a SendTx).

**Comment by ebuchman** [comment]
ok I'm actually not sure about that last commit, since if you create a contract and want to talk to it in the same block, a proposer can just reverse the order of your transactions and make one invalid but still collect fees. What do you think? Should we just allow CallTx's to empty addresses as like a cheap send tx?

**Comment by ebuchman** [comment]
oo so actually it's fine if I'm one person because of the nonce but not if we're two people. Like if you deploy a contract and I try and talk to it in the same block, block makers can screw us by putting my transaction first and taking some fees.

**Comment by ebuchman** [comment]
maybe its too remote a consideration and we can just say "if you're more than one person you'll have to wait a block before talking to a new contract or else you could lose some fees" lol

File: vm/vm.go [hunk 1]
```diff
@@ -71,38 +71,43 @@ func (vm *VM) SetFireable(evc events.Fireable) {
...
 
-	if len(code) == 0 {
-		panic("Call() requires code")
-	}
+	exception := new(string)
+	defer func() {
+		// if callDepth is 0 the event is fired from ExecTx (along with the Input event)
+		// otherwise, we fire from here.
+		if vm.callDepth != 0 && vm.evc != nil {
+			fmt.Println("FIRE AWAY!", types.EventStringAccReceive(callee.Address.Postfix(20)))
```
Comment (ebuchman) [inline vm/vm.go]: remove this println

**Pull request merged**